lib/MadNLPGPU/src/cudss.jl

```diff
@@ -0,0 +1,98 @@
+import CUDSS
+import SparseArrays
+
+@kwdef mutable struct CudssSolverOptions <: MadNLP.AbstractOptions
+    # Use LDL by default in CUDSS as Cholesky can lead to undefined behavior. 
+    cudss_algorithm::MadNLP.LinearFactorization = MadNLP.LDL
+end
+
+mutable struct CUDSSSolver{T} <: MadNLP.AbstractLinearSolver{T}
+    inner::Union{Nothing, CUDSS.CudssSolver}
+    tril::CUSPARSE.CuSparseMatrixCSC{T}
+    x_gpu::CUDA.CuVector{T}
+    b_gpu::CUDA.CuVector{T}
+
+    opt::CudssSolverOptions
+    logger::MadNLP.MadNLPLogger
+end
+
+function CUDSSSolver(
+    csc::CUSPARSE.CuSparseMatrixCSC{T};
+    opt=CudssSolverOptions(),
+    logger=MadNLP.MadNLPLogger(),
+) where T
+    n, m = size(csc)
+    @assert n == m
+
+    view = 'U'
```

Comment:
If the input is `tril`, why are you not using `view = 'L'` ?

Reply:
We just convert CSC into CSR (by not changing anything, but renaming colptr to rowptr and rowval to coeval). So, it is upper triangular

Reply:
Oh, I see. Thanks for the explanation!

Note that `view = 'F'` is only possible value for LU decomposition.
It could be great to allow the LU decomposition of cuDSS (even if `A` is symmetric) because it will contain the most robust algorithm in the next release of cuDSS (0.2.0).
I can open a PR for that.

cc @frapac

Reply:
I see. Indeed it would be beneficial to allow using LU. Thanks, @amontoison!